python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -849,7 +850,15 @@ def _raise_on_error(self) -> None:
 T = TypeVar("T", bound=DbtDagsterEventType)
 
 
-class DbtEventIterator(abc.Iterator[T]):
+# In 3.8, the Iterator type is not generic, but we can instead implement typing.Iterator
+BaseIterator = abc.Iterator
```

Comment:
why the initial set?

Reply:
ended up changing approach to use an un-parameterized `Iterator`+ `Generic[T]`, that seems to satisfy editor type hinting + pyright + 3.8